### ISSUE TITLE
Feature Request: Support for --build-arg in docker task

### ISSUE THREAD

**Issue by rmelick** [palantir/gradle-docker#45]
I'm attempting to build a docker image that contains a versioned tar (kiwiCloud-6.0.0.0-SNAPSHOT.tar) file that is produced by a different part of my build.

build.gradle file

```
plugins {
    id 'com.palantir.docker' version '0.7.0'
}

docker {
    name "docker.com/rmelick/cloud/api:$version"
    dependsOn tasks.getByPath(':distTar')
}

```

Dockerfile

```
FROM java:8

ADD kiwiCloud-* /
```

I would like to be able to pass the version parameter into my Dockerfile as a build-arg.  This would let me remove the wildcard from the ADD command, and would also let me use the version when referencing the contents of the tar (the script I want to use for ENTRYPOINT is inside the tar).

**Comment by markelliot** [comment]
A couple of ways you could do this:
- use Docker arguments (https://docs.docker.com/engine/reference/builder/#arg)
- use Gradle substitution to generate your Dockerfile (and configure this plugin to point at the generated file)

I'm not sure it makes sense for this plugin to become involved in modifying your Dockerfile, though.

**Comment by rmelick** [comment]
@markelliot I agree, I think using the docker --build-arg argument would be the right way to go, rather than modifying the Dockerfile.  I'll try to send a PR today.

**Comment by markelliot** [comment]
Released in 0.7.1
